apache/superset
# fix(dashboard): Charts crashing when cross filter on adhoc column is applied

### SUMMARY
When there are charts using different datasets in a dashboard and a cross filter is applied on an adhoc column, the charts that use different dataset than the origin chart are crashing with a db error.
This PR fixes that by excluding filters based on an incompatible adhoc column from the query.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/15073128/221936633-0af12308-5dc3-4e8f-ae63-acdb6addcfee.png">

After:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/15073128/221937754-bfa8a068-1d0c-4bf7-a1d0-fb14dba83c04.png">

### TESTING INSTRUCTIONS
1. Create a chart that is grouped by an adhoc column (with sql expression) and add it to a dashboard
2. Add another chart to a dashboard, which should use a different dataset than the first chart
3. Apply a cross filter by clicking on a series related to the adhoc column in the first chart
4. Verify that the second chart doesn't crash and is not filtered by the cross filter

### ADDITIONAL INFORMATION
